### PR TITLE
Update wallet generation error UI

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/38968

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java [hunk 1]
```diff
@@ -1827,6 +1827,7 @@ private void showRewardsResponseModal(boolean isSuccess, String errorMessage) {
             String title = mActivity.getString(R.string.something_went_wrong_text);
             String text = mActivity.getString(R.string.declare_geo_failed_response_text);
             String actionText = mActivity.getString(R.string.retry_text);
+            Log.e("wallet_error", "BraveRewardsPanel : errorMessage : "+errorMessage);
```
Comment (tapanmodh) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:1830]: do we need to log? or is it leftover?

File: android/java/org/chromium/chrome/browser/rewards/onboarding/RewardsOnboarding.java [hunk 2]
```diff
@@ -260,9 +261,11 @@ private void updateErrorModalDescription(String errorMessage) {
         TextView responseModalText =
                 mPopupView.findViewById(R.id.rewards_onboarding_error_description);
         TextView responseRewardsBtn = mPopupView.findViewById(R.id.response_action_btn);
+        TextView responseModalTitle = mPopupView.findViewById(R.id.rewards_onboarding_error_title);
         TextView responseErrorText = mPopupView.findViewById(R.id.response_error_text);
 
         String actionText = mPopupView.getContext().getString(R.string.retry_text);
+        Log.e("wallet_error", "RewardsOnboarding : errorMessage : "+errorMessage);
```
Comment (tapanmodh) [inline android/java/org/chromium/chrome/browser/rewards/onboarding/RewardsOnboarding.java:268]: do we need to log? or is it leftover?

**Review by tapanmodh — APPROVED** [review]
lgtm

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on `Vivo X70 Pro version 12` running `Bravemonoarm64.apk_1.69.64`

- Verified the description from the issue https://github.com/brave/brave-browser/issues/38968#issue-2347030266 and ensured that the following text is shown completely on the BR panel. 
- Confirmed that the text is matching with the Desktop message
- Confirmed that user view the complete message on the panel there is no text is cutoff in the BR panel
>Brave Rewards not available
New signups for Brave Rewards are currently disabled in your region. However, you can always try again later. Learn more

Screenshot_20240628_202732

https://github.com/brave/brave-core/assets/38657976/11dc4b5b-5534-4f48-823a-c73335505212

**Pull request merged**